### PR TITLE
setting up docker for backend

### PR DESCRIPTION
**Pull request by Muhet**
#### What does this PR do?
setup docker compose for our backend project

#### Description of Task to be completed?
As a developer, I want to setup  docker so that new images can be created for deployment.

```
How Can I test it


git clone https://github.com/atlp-rwanda/atlp-apprenticeship-bn.git
cd atlp-apprenticeship-bn
git checkoutch-settingUp-Docker
npm install

Intall Docker desktop in your PC and make sure all the settings is done and you can be able to open it.

Go back to your VS code terminal run the following commands

1. docker build -t <<the name of the image>> . 
2. docker image ls  /* for listing all image that you have */
3. docker run -p 4000:4000 -d --name   /*the name of the container*/   /*name of the image*/
4. docker ps  /*This is to see the container opened at this moment*/
5.  Go to your Browser and run localhost:4000
You should be able to run server




```

#### What are the relevant pivotal tracker/Trello stories?

Trello [Link]() (#-https://trello.com/c/axA4L3dX)

#### Screenshots (if appropriate)  )

||
ApprenticeshipTwo
||
Docker
||
DockerTwo

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM node:slim
```
Comment (ceelogre) [inline Dockerfile]: What's the difference between slim and latest?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+WORKDIR /app
+COPY . /app
+RUN npm install
```
Comment (ceelogre) [inline Dockerfile]: How comes there is no build step?

**Review by ceelogre — CHANGES_REQUESTED** [review]

**Review by rukundo-kevin — APPROVED** [review]

**Review by Virgile1k — APPROVED** [review]
Checked and approved

**Review by Lydie260 — APPROVED** [review]

**Review by NTElissa — APPROVED** [review]

**Review by NTElissa — APPROVED** [review]

**Review by ceelogre — CHANGES_REQUESTED** [review]
Merging develop into your feature branch is not accepted. Please reset this commit.
image

**Pull request closed without merging**